packages/libirmin/libirmin.3.4.3/opam

```diff
@@ -0,0 +1,33 @@
+opam-version: "2.0"
+synopsis: "C bindings for irmin"
+description: "C bindings for irmin using Ctypes inverted stubs"
+maintainer: ["zachshipko@gmail.com"]
+authors: ["Zach Shipko"]
+license: "ISC"
+homepage: "https://github.com/mirage/irmin"
+bug-reports: "https://github.com/mirage/irmin/issues"
+depends: [
+  "dune" {>= "2.9"}
+  "ctypes" {>= "0.19"}
+  "ctypes-foreign" {>= "0.18"}
+  "irmin" {= version}
+  "irmin-cli" {= version}
+]
+build: [
+  ["dune" "subst"] {dev}
+  ["dune" "build" "-p" name "-j" jobs]
+  ["dune" "runtest" "-p" name "-j" jobs] {with-test}
+]
+dev-repo: "git+https://github.com/mirage/irmin.git"
+
+# Disabled on arm64 linux and s390x because of a SEGFAULT in tests
+available: [ (arch != "arm64" | os = "macos") & arch != "s390x" ]
```

Comment:
To have a full explanation, it's weird that your `generate.exe` used to generate few files needed to compile `libirmin` is linked with `ca-certs`. It seems that the produced executable then try to have an access to trust anchors and fail on MacOS.

Reply:
thanks @kit-ty-kate for the suggestion, I disabled macos on several irmin subpackages that had the same error. 

@dinosaure I'm not sure where this error is coming from.

Reply:
@dinosaure - I don't think it's specific to `generate.exe`, all of the `irmin` packages that depend on `conduit-lwt-unix` have this issue. Is there any workaround?

Reply:
Currently, the error is much more related to MacOS which has a bad sandboxing. @kit-ty-kate already sent an issue _upstream_.  What appears in this error is that your program needed to **build** `libirmin` (we are not talking about tests or artifacts produced) is linked to `ca-certs`. The latter will then try to access the trust anchors and the sandboxing of MacOS fails at this level.

The log is lost but I have seen that `generate.exe` used to build `libirmin` is linked with `ca-certs` (not because it depends on it explicitly but it appears in the build command) which is really strange (when you look at what this program actually does).

I don't have a Mac so I can't reproduce the error or introspect further unfortunately.

Reply:
I suspect indeed this line but again, I can not really help 🙂 .

Reply:
FWIW, PRs welcome to use the macOS API (esp. those who have their hands on a macOS computer could do so), see https://github.com/mirage/ca-certs/issues/20 (including a link to the go implementation).

Reply:
Thanks for the pointer, I will take a look!